android/src/main/kotlin/com/kasem/media_picker_builder/providers/VideoFileProvider.kt

```diff
@@ -2,93 +2,189 @@ package com.kasem.media_picker_builder.providers
+                                mediaFile.albumName,
+                                mutableListOf(mediaFile)
+                        )
+                    } else {
+                        album.files.add(mediaFile)
+                    }
+                }
+            }
+        } else {
+            val cursor = context.contentResolver.query(
+                    MediaStore.Video.Media.EXTERNAL_CONTENT_URI,
+                    arrayOf(
+                            MediaStore.Video.Media._ID,
+                            MediaStore.Video.Media.DATE_ADDED,
+                            MediaStore.Video.Media.DATA,
+                            MediaStore.Video.Media.MIME_TYPE),
+                    selectionClause,
+                    selectionArgs,
+                    "${MediaStore.Video.Media._ID} DESC")
+
+            val mediaMetadataRetriever = MediaMetadataRetriever()
+            while (cursor?.moveToNext() == true) {
+                val mediaFile = cursorToMediaFilePriorTo29(mediaMetadataRetriever, cursor)
+
+                if (mediaFile != null) {
+                    val album = albumHashMap[mediaFile.albumId]
+                    if (album == null) {
+                        albumHashMap[mediaFile.albumId] = Album(
+                                mediaFile.albumId,
+                                mediaFile.albumName,
+                                mutableListOf(mediaFile)
+                        )
+                    } else {
+                        album.files.add(mediaFile)
+                    }
                 }
             }
+            mediaMetadataRetriever.release()
         }
     }
 
-    private fun cursorToMediaFile(cursor: Cursor): MediaFile {
-        val fileId = cursor.getLong(0)          //MediaStore.Video.Media._ID
-        val fileDateAdded = cursor.getLong(1)   //MediaStore.Video.Media.DATE_ADDED
-        val filePath = cursor.getString(2)      //MediaStore.Video.Media.DATA
-        val albumName = cursor.getString(3)     //MediaStore.Video.Media.BUCKET_DISPLAY_NAME
-        val albumId = cursor.getLong(4)         //MediaStore.Video.Media.BUCKET_ID
-        val mimeType = cursor.getString(5)      //MediaStore.Video.Media.MIME_TYPE
-        val duration = cursor.getLong(6) / 1000 //MediaStore.Video.Media.DURATION
-        val orientation = cursor.getInt(7)      //MediaStore.Video.Media.ORIENTATION
+
+    private fun cursorToMediaFileSubsequent29(cursor: Cursor): MediaFile? {
+        val fileId = cursor.getLong(0)
+        val fileDateAdded = cursor.getLong(1)
+        val filePath = cursor.getString(2)
+        val mimeType = cursor.getString(3)
+        val albumId = cursor.getLong(4)
+        val duration = cursor.getDouble(5) / 1000
+        val orientation = cursor.getInt(6)
 
         return MediaFile(
                 fileId,
                 albumId,
-                "", //Passing empty string, because real `albumName` was problematic under certain circumstances
+                "",
```

Comment:
☝️ lets keep that comment here for now.

Reply:
Non needed anymore as albumName is not properly taken accoding to android sdk version